roxie/ccd/ccdcontext.cpp

```diff
@@ -3497,11 +3497,11 @@ class CRoxieServerContext : public CRoxieContextBase, implements IRoxieServerCon
     {
         CriticalBlock b(contextCrit);
         StringBuffer expandedName;
-        expandLogicalFilename(expandedName, fileName, workUnit, false, !workUnit);
+        expandLogicalFilename(expandedName, fileName, workUnit, false, !serverQueryFactory->isDynamic());
```

Comment:
That's a tricky one.  Seems like it would get confusing either way.  For consistency I might error on the side of moving the DYNAMIC to the local DALI just as we did the foreign.. but I've no problem with doing the opposite if you preferred it.